### PR TITLE
Fixes #3560  Merge profile_list_fragment.xml into single xml file

### LINKED ISSUE #3560: Merge profile_list_fragment.xml into single xml file
Currently there are 3 versions of `profile_list_fragment.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by Rishavgupta12345**
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Fix #bugnum: #3560   Merge profile_list_fragment.xml into single xml file

screenshorts :-

mobile portrait before -
Screenshot 2022-01-10 114723

mobile portrait after -
Screenshot 2022-01-10 114723

mobile landscape before -
Screenshot 2022-01-10 114821

mobile landscape after -
Screenshot 2022-01-10 114821

tab portrait before -
Screenshot 2022-01-10 115203

tab portrait after -
Screenshot 2022-01-10 115203

tab landscape before -
Screenshot 2022-01-10 115227

tab landscape after -
Screenshot 2022-01-10 115227

### CODE REVIEW

**Comment by Rishavgupta12345** [comment]
PTAL @rt4914

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]
@Rishavgupta12345 the screenshots you have provided is incorrect, you have to run the app in your mobile or emulator and then take screenshots

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 the screenshots you have provided is incorrect, you have to run the app in your mobile or emulator and then take screenshots

then I have to download many emulators , for mobile and tab all of that

**Comment by Rishavgupta12345** [comment]
PTAL @rt4914

**Comment by bkaur-bkj** [comment]
> > @Rishavgupta12345 the screenshots you have provided is incorrect, you have to run the app in your mobile or emulator and then take screenshots
> 
> then I have to download many emulators , for mobile and tab all of that

yes you need to download a mobile and tablet emulator

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]
@Rishavgupta12345 your branch has merge conflicts, kindly resolve them

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 your branch has merge conflicts, kindly resolve them

done resolved

**Comment by bkaur-bkj** [comment]
@Rishavgupta12345 remember to keep the assignee section updated, if you wont assign us back we won't be notified

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -590,7 +590,14 @@
 
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">132dp</dimen>
+
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values/dimens.xml]: @Rishavgupta12345 why are extra line spaces here?

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -590,7 +590,14 @@
...
+  <!-- ProfileListFragment -->
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values/dimens.xml:598]: @Rishavgupta12345 also here?

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]
@Rishavgupta12345to resolve conflicts you don't need to add these extra blank lines, kindly remove them, if you know that the conflict which is coming is fine or can be pushed so remove that comment line from code and push it back after marking it resolved

**Comment by Rishavgupta12345** [comment]
PTAL @bkaur-bkj done the requested changes

**Comment by Rishavgupta12345** [comment]
PTAL @bkaur-bkj

File: app/src/main/res/values-land/dimens.xml [hunk 3]
```diff
@@ -392,7 +392,14 @@
 
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">96dp</dimen>
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values-land/dimens.xml:399]: @Rishavgupta12345 As I asked in last review these are not removed

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]

**Comment by bkaur-bkj** [comment]
@Rishavgupta12345 also you don't need to resolve any conversation yourself, just reply done to my msg

File: app/src/main/res/values-land/dimens.xml (hunk 3 above)
Comment (Rishavgupta12345) [reply app/src/main/res/values-land/dimens.xml:399]: done PTAl @bkaur-bkj

**Comment by Rishavgupta12345** [comment]
PTAL @bkaur-bkj

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]
@Rishavgupta12345 the fragments are not merged properly , we have app bar in tablet mode whereas we don't have that in mobile layout , also we have padding bottom va;lue different also which is not addressed kindly use textcompare to compare the differences between files

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 the fragments are not merged properly , we have app bar in tablet mode whereas we don't have that in mobile layout , also we have padding bottom va;lue different also which is not addressed kindly use textcompare to compare the differences between files

that app bar in tablet mode is a layout configuration of the tablet emulater i am using . It has nothing to do with the code . if someone use different emulater for that then he wont get that app bar , nothing to worry with it PTAL @bkaur-bkj

File: app/src/main/res/layout-land/profile_list_fragment.xml [hunk 4]
```diff
@@ -1,70 +0,0 @@
...
-
-  <androidx.constraintlayout.widget.ConstraintLayout
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@color/white">
-
-    <com.google.android.material.appbar.AppBarLayout
-      android:id="@+id/profile_list_app_bar_layout"
-      android:layout_width="0dp"
-      android:layout_height="wrap_content"
```
Comment (Arjupta) [inline app/src/main/res/layout-land/profile_list_fragment.xml:20]: > that app bar in tablet mode is a layout configuration of the tablet emulater i am using . It has nothing to do with the code . if someone use different emulater for that then he wont get that app bar , nothing to worry with it PTAL @bkaur-bkj

@Rishavgupta12345 I see that you have merged the sw600dp (tablet) and mobile xml files without considering the fact that the tablet one doesn't have any `AppBarLayout` component. This means when we would be using `profile_list_fragment` for a tablet device, we will be seeing an extra AppBar in the screen (You can see that there is an unnecessary AppBar shown in the tablet screenshots)

File: app/src/main/res/values-land/dimens.xml [hunk 5]
```diff
@@ -392,7 +392,11 @@
 
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">96dp</dimen>
-  
+
+  <!-- ProfileListFragment -->
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
+
   <!-- StoryFragment -->
```
Comment (Arjupta) [inline app/src/main/res/values-land/dimens.xml:403]: The `profile_list_fragment_recycler_view_padding_bottom` for the landscape xml file is 108dp, please verify this at your end.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 6]
```diff
@@ -399,4 +399,7 @@
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">132dp</dimen>
 
+  <!-- ProfileListFragment -->
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
+
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-land/dimens.xml:409]: As the `profile_list_fragment_recycler_view_padding_bottom` for both sw600dp and mobile layout is same (40dp) therefore we don't need to create another dimens here.

File: app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -590,7 +590,11 @@
...
   <dimen name="audio_language_recycler_view_padding_bottom">132dp</dimen>
-   
+
+  <!-- ProfileListFragment -->
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
+
   <!-- StoryFragment -->
   <dimen name="story_fragment_padding_bottom">160dp</dimen>
+
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values/dimens.xml]: Avoid introducing unnecessary blank lines. Ditto in other xml files too.

**Review by Arjupta — CHANGES_REQUESTED** [review]
@Rishavgupta12345 PTAL at the requested changes and reply done to all of them once you have committed your changes. Also avoid resolving the comments on your own.

File: app/src/main/res/values/dimens.xml (hunk 7 above)
Comment (Rishavgupta12345) [reply app/src/main/res/values/dimens.xml]: this isn't a extra line basically after line 598 a gap of line 599 was already present to separate with the line 600 , so it is a necessary thing to do

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 6 above)
Comment (Rishavgupta12345) [reply app/src/main/res/values-sw600dp-land/dimens.xml:409]: so should i delete this three lines ???

File: app/src/main/res/values-land/dimens.xml (hunk 5 above)
Comment (Rishavgupta12345) [reply app/src/main/res/values-land/dimens.xml:403]: yes I am rectifying it

File: app/src/main/res/layout-land/profile_list_fragment.xml (hunk 4 above)
Comment (Rishavgupta12345) [reply app/src/main/res/layout-land/profile_list_fragment.xml:20]: how should I resolve this problem , can you please guide me

**Review by Arjupta — CHANGES_REQUESTED** [review]
@Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments

I have written a reply to each of your suggested comments I just want you to give a look at them

**Comment by Rishavgupta12345** [comment]
> > @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments
> 
> I have written a reply to each of your suggested comments I just want you to give a look at them and guide me as requested

**Comment by rt4914** [comment]
> > @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments
> 
> I have written a reply to each of your suggested comments I just want you to give a look at them

@Rishavgupta12345 Can you please check that again as I am unable to see your replies on previous comments.

**Comment by Rishavgupta12345** [comment]
> > > @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments
> > 
> > 
> > I have written a reply to each of your suggested comments I just want you to give a look at them
> 
> @Rishavgupta12345 Can you please check that again as I am unable to see your replies on previous comments.

GO TO THE FILES CHANGED OPTION you will get the comments there @ARJUPTA @rt4914

**Comment by rt4914** [comment]
@Rishavgupta12345 

> > > @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments
> > 
> > 
> > I have written a reply to each of your suggested comments I just want you to give a look at them
> 
> @Rishavgupta12345 Can you please check that again as I am unable to see your replies on previous comments.

@Rishavgupta12345 Actually it will be visible to you only and not us. You have commented in "Files changed" section in that case to submit all your comments for everyone you will need to "Review changes" green-button on top-right corner and click on "Submit review". Currently the changes are available on your local machine only and have not been pushed for public view.
Screenshot 2022-02-02 at 1 18 40 PM

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345
> 
> > > > @Rishavgupta12345 you have not made any of the suggested changes. If you feel the suggested changes are not reasonable then reply with an explaination for the same so that we can discuss it over comments
> > > 
> > > 
> > > I have written a reply to each of your suggested comments I just want you to give a look at them
> > 
> > 
> > @Rishavgupta12345 Can you please check that again as I am unable to see your replies on previous comments.
> 
> @Rishavgupta12345 Actually it will be visible to you only and not us. You have commented in "Files changed" section in that case to submit all your comments for everyone you will need to "Review changes" green-button on top-right corner and click on "Submit review". Currently the changes are available on your local machine only and have not been pushed for public view. Screenshot 2022-02-02 at 1 18 40 PM

acha got it now

File: app/src/main/res/values-land/dimens.xml [hunk 8]
```diff
@@ -396,11 +396,15 @@
...
   <dimen name="story_fragment_padding_bottom">60dp</dimen>
 
   <!-- TopicLessonsTitle -->
   <dimen name="topic_lessons_title_margin_start">100dp</dimen>
   <dimen name="topic_lessons_title_margin_end">100dp</dimen>
+
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:409]: Remove this extra blank line

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -592,12 +592,16 @@
...
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
+
   <!-- StoryFragment -->
   <dimen name="story_fragment_padding_bottom">160dp</dimen>
 
   <!-- TopicLessonsTitle -->
   <dimen name="topic_lessons_title_margin_start">32dp</dimen>
   <dimen name="topic_lessons_title_margin_top">32dp</dimen>
   <dimen name="topic_lessons_title_margin_end">32dp</dimen>
+
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:606]: Remove this extra blank line

File: app/src/main/res/values-land/dimens.xml [hunk 10]
```diff
@@ -396,11 +396,15 @@
   
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">96dp</dimen>
-  
+
+  <!-- ProfileListFragment -->
+  <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:401]: This value should be `108dp` please check and confirm this once again.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Rishavgupta12345 You cannot make changes to tablet file because if you compare it correctly the tablet file does not contain toolbar and the mobile related files do contain toolbar.

Undo the changes file tablet files.

**Comment by rt4914** [comment]
@Rishavgupta12345 @ARJUPTA Any updates on this?

**Comment by Arjupta** [comment]
> @Rishavgupta12345 @ARJUPTA Any updates on this?

@rt4914 I guess @Rishavgupta12345 is inactive currently so we can close this PR now so that any other contributor can work in this issue.

**Comment by rt4914** [comment]
Thanks @ARJUPTA . Make the necessary adjustments to new-contributor-tracker.

**Pull request closed without merging**